### ISSUE TITLE
[Tests] LayerNormTestSet platform applicability issue after #2481

### ISSUE THREAD

**Issue by junliume** [ROCmSoftwarePlatform/MIOpen#2527]
After #2481 some CI stages start to fail when the tests are performed on legacy platforms:

For example, on `gfx906`:
```
The following tests FAILED:
	941 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:64 C:4 D:0 H:4 W:256 dim:1 eps:1e-05 LayerNorm_mode:0 (Failed)
	943 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:64 C:4 D:0 H:4 W:256 dim:1 eps:1e-05 LayerNorm_mode:1 (Failed)
	945 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:64 C:0 D:0 H:0 W:256 dim:1 eps:1e-05 LayerNorm_mode:0 (Failed)
	947 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:64 C:0 D:0 H:0 W:256 dim:1 eps:1e-05 LayerNorm_mode:1 (Failed)
```

The error message:
```
unknown file: Failure
C++ exception with description "HIP runtime error: invalid device function. /opt/rocm/cget/build/tmp-6a54a0c103a88df9ce804b69df353620d496a4a26d329eb22384908711114873/composable_kernel-f30e59754205db83eea8e4ea42caf421eafb70e0/include/ck/host_utility/hip_check_error.hpp: 15in function: hip_check_error" thrown in TearDown().
```

We should **protect** and be **explicit** about test applicabilities specific to platforms, i.e. only run these tests on applicable ASICs.

**Comment by iq136boy** [comment]
@junliume this could be done by setting the platforms/device list in the TEST_P function in gtest. Sth like the following: 
    const auto& handle = get_handle();
    if(!((miopen::StartsWith(handle.GetDeviceName(), "gfx908") ||
        miopen::StartsWith(handle.GetDeviceName(), "gfx90a")))    // set the device list for the gtest
    {
        GTEST_SKIP();
    }

**Comment by atamazov** [comment]
@junliume 
> After https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2481 some CI stages start to fail

❓ How is this possible if PR has passed our CI?

If this is really so, then our testing procedure (defined in Jenkinsfile) has serious flaws that must be fixed.

**Comment by junliume** [comment]
> @junliume
> 
> > After #2481 some CI stages start to fail
> 
> ❓ How is this possible if PR has passed our CI?
> 
> If this is really so, then our testing procedure (defined in Jenkinsfile) has serious flaws that must be fixed.

Several stages are marked as "anyGPU" so it could happen that by chance all stages were performed on gfx908;gfx90a

**Comment by junliume** [comment]
@atamazov @JehandadKhan @seungmanhan  The issue is still not resolved even after #2528
```
 941/1601 Test  #947: LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:64 C:0 D:0 H:0 W:256 dim:1 eps:1e-05 LayerNorm_mode:1 ...................................................................................................................................................................................................................................................................***Failed    0.64 sec
Running main() from /opt/rocm/cget/build/tmp-c4f1a42bcee536579a1248d25169f834201170c187fc6d80e8c510a7d28f6f48/googletest-1.14.0/googletest/src/gtest_main.cc
Note: Google Test filter = LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/65
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from LayerNormTestSet/LayerNormTestFloat
[ RUN      ] LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/65
MIOpen(HIP): Info [get_device_name] Raw device name: gfx906:sramecc+:xnack-
MIOpen(HIP): Info [Handle] stream: 0x20bea90, device_id: 3
PRNG seed: 12345678
/var/jenkins/workspace/MLLibs_MIOpen_develop/test/gtest/layernorm.cpp:50: Skipped


MIOpen(HIP): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv3, HIP version 5.7.23384, MIOpen version 3.00.0.5a1a840ce
unknown file: Failure
C++ exception with description "HIP runtime error: invalid device function. /opt/rocm/cget/build/tmp-6a54a0c103a88df9ce804b69df353620d496a4a26d329eb22384908711114873/composable_kernel-f30e59754205db83eea8e4ea42caf421eafb70e0/include/ck/host_utility/hip_check_error.hpp: 15in function: hip_check_error" thrown in TearDown().
```

It is strange that even though the test is skipped, it still emits errors. Hence all other `LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw` were skipped properly:
```
	882 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:32 C:1 D:32 H:32 W:32 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	883 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:32 C:1 D:14 H:14 W:14 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	884 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:32 C:32 D:14 H:14 W:14 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	885 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:32 C:32 D:12 H:12 W:12 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	886 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:32 C:32 D:6 H:6 W:6 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	887 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:256 C:1 D:32 H:32 W:32 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	888 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:256 C:32 D:14 H:14 W:14 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	889 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:256 C:32 D:12 H:12 W:12 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	890 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:256 C:32 D:6 H:6 W:6 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	891 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:512 C:1 D:32 H:32 W:32 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	892 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:512 C:32 D:14 H:14 W:14 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)

	893 - LayerNormTestSet/LayerNormTestFloat.LayerNormTestFw/ N:512 C:32 D:12 H:12 W:12 dim:4 eps:1e-05 LayerNorm_mode:0 (Skipped)
```


What is so special about `N;64`?

**Comment by seungmanhan** [comment]
I checked and it seems that the kernel is executed and skipped. I will upload a modified PR for the layernorm test.
